_posts/2021-10-25-waku-v1-vs-waku-v2.md

```diff
@@ -0,0 +1,202 @@
+From time to time, a Waku v2 peer will send metadata about the messages it has seen to other peers ("gossip" peers).
+
+See [this explainer](https://hackmd.io/@vac/main/%2FYYlZYBCURFyO_ZG1EiteWg#11WAKU2-RELAY-gossipsub) for a more detailed discussion.
+
+## Methodology
+
+The results below contain only some scenarios that provide an interesting contrast between Waku v1 and Waku v2.
+For example, [star network topologies](https://www.techopedia.com/definition/13335/star-topology#:~:text=Star%20topology%20is%20a%20network,known%20as%20a%20star%20network.) do not show a substantial difference between Waku v1 and Waku v2.
+This is because each peer relies on a single connection to the central node for every message,
+which barely requires any routing:
+each connection receives a copy of every message for both Waku v1 and Waku v2.
+Hybrid topologies similarly show only a difference between Waku v1 and Waku v2 for network segments with [mesh-like connections](https://en.wikipedia.org/wiki/Mesh_networking),
+where routing decisions need to be made.
+
+For this reason, the following approach applies to all scenarios:
+1. Simulations are run **locally**.
+This limits the size of possible scenarios due to local resource constraints,
+but is a way to quickly get an approximate comparison.
+2. Nodes are treated as a **blackbox** for which we only measure bandwidth,
+using an external bandwidth monitoring tool.
+In other words, we do not consider differences in the size of the envelope (for v1) or the message (for v2).
+3. Messages are published at a rate of **50 new messages per second** to each network,
+except where explicitly stated otherwise.
+4. Each message propagated in the network carries **8 bytes** of random payload, which is **encrypted**.
+The same symmetric key cryptographic algorithm (with the same keys) are used in both Waku v1 and v2.
+5. Traffic in each network is **generated from 10 nodes** (randomly-selected) and published in a round-robin fashion to **10 topics** (content topics for Waku v2).
+In practice, we found no significant difference in _average_ bandwidth usage when tweaking these two parameters (the number of traffic generating nodes and the number of topics).
+6. Peers are connected in a decentralized **full mesh topology**,
+i.e. each peer is connected to every other peer in the network.
+Waku v1 is expected to flood all messages across all existing connections.
+Waku v2 gossipsub will GRAFT some of these connections for full-message peerings,
+with the rest being gossip-only peerings.
+7. After running each scenario, we **verify that messages propagated to all peers** (comparing the number of published messages to the metrics logged by each peer).
+
+For Waku v1, nodes are configured as "full" nodes (i.e. with full bloom filter),
+while Waku v2 nodes are `relay` nodes, all subscribing and publishing to the same PubSub topic.
+
+## Network size comparison
+
+### Scenario 1: 10 nodes
```

Comment:
Nitpick but, scenario to me indicates substantially different. Here we are (mostly, from what I can tell) changing one variable. Scenario I'd expect to vary topology or fundamental usage.

Maybe a better word exist? If not, can keep it

Reply:
In fact, this makes sense in terms of how you discovered and ran for different N. But if we "start from scratch", maybe the scenarios can be compacted to something like "a run" varying number of nodes? This would make it more concise, and then discussion/analysis would point to ~linear relationship.

Reply:
If this seems like too much work don't have to do it, just an idea

Reply:
Agree. In 21749a4 I changed these to "iterations" over the same scenario with each iteration changing the network size. LMK if you think this is a better term. "Scenario" still works in some contexts, so only changed where it made sense.